src/blue-magic/blue-magic-caster.cpp

```diff
@@ -93,20 +93,21 @@ static bool cast_blue_hand_doom(PlayerType *player_ptr, bmc_type *bmc_ptr)
     return true;
 }
 
-static std::pair<bool, std::string> exe_blue_teleport_back(PlayerType *player_ptr)
+/* 効果が抵抗された場合、返される std::optional には値がありません。*/
+static std::optional<std::string> exe_blue_teleport_back(PlayerType *player_ptr)
 {
     MonsterEntity *m_ptr;
     MonsterRaceInfo *r_ptr;
     auto *floor_ptr = player_ptr->current_floor_ptr;
     if ((floor_ptr->grid_array[target_row][target_col].m_idx == 0) || !player_has_los_bold(player_ptr, target_row, target_col) || !projectable(player_ptr, player_ptr->y, player_ptr->x, target_row, target_col)) {
-        return std::make_pair(true, std::string());
+        return std::optional<std::string>();
```

Comment:
You can use std::nullopt instead, which is clearer.

Reply:
The forced push should have std::nullopt for the returns that used std::optional<std::string>().